sw/device/lib/dif/dif_keymgr.h

```diff
@@ -0,0 +1,648 @@
+ * key manager relies on the entropy source to initialize its secret value.
+ *
+ * This function must be called only once per power cycle.
+ *
+ * @param keymgr A key manager handle.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_keymgr_result_t dif_keymgr_initialize_state(const dif_keymgr_t *keymgr);
+
+/**
+ * Parameters for advancing the state of key manager.
+ */
+typedef struct dir_keymgr_advance_state_params {
+  /**
+   * This value is used by key manager to derive secret values and can be either
+   * a value that represents the contents of a boot stage, e.g. a truncated
+   * hash, or a tag. If it is a hash, changes in a boot stage will change the
+   * secret value, and consequently the versioned keys and identity seeds
+   * generated at subsequent boot stages. If it is a tag, those secret values,
+   * versioned keys, and identity seeds will be preserved across updates of the
+   * boot stage as long as the tag remains the same.
+   */
+  uint32_t sw_binding[DIF_KEYMGR_SW_BINDING_WORDS];
+  /**
+   * Maximum allowed version for keys generated at the next state.
+   */
+  uint32_t max_key_version;
+  /**
+   * Expected next state of the key manager. This value is used only for
+   * error checking.
+   */
+  dif_keymgr_state_t expected_next_state;
+} dif_keymgr_advance_state_params_t;
+
+/**
+ * Advances key manager state.
+ *
+ * This function sets the max version value and the software binding value that
+ * is used to compute the secret value for the next state of key manager and
+ * initiates a transition to the next state, i.e. Initialized -> CreatorRootKey
+ * -> OwnerIntermediateKey -> OwnerRootKey -> Disabled. State transitions are
+ *  irreversible during a power cycle. Key manager must be initialized using
+ *  `dif_keymgr_initialize_state()` prior to calling this function.
+ *
+ * This is an asynchronous function because key manager state transitions
+ * involve KMAC operations that can take some time to complete. Clients must
+ * check the status of key manager using `dif_keymgr_get_status()` before
+ * calling other functions in this library.
+ *
+ * This function also checks that key manager is in the correct state for the
+ * expected next state.
+ *
+ * @param keymgr A key manager handle.
+ * @param params State transition parameters.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_keymgr_lockable_result_t dif_keymgr_advance_state(
+    const dif_keymgr_t *keymgr, dif_keymgr_advance_state_params_t params);
+
+/**
+ * Disables key manager.
+ *
+ * This function disables key manager until the next power cycle by making it
+ * transition to Disabled state. Disabled state is a terminal state where key
+ * manager is no longer operational and its secret value is a random value.
+ *
+ * @param keymgr A key manager handle.
+ * @return The result of the operation.
+ */
+DIF_WARN_UNUSED_RESULT
+dif_keymgr_lockable_result_t dif_keymgr_disable(const dif_keymgr_t *keymgr);
```

Comment:
This should be called `dif_keymgr_lock()`, I think?

(With an `is_locked()` companion.)

Reply:
I considered this option too, but chose this name to mostly align with the tech spec. Also, since we already have a getter for the status, I thought is_locked would be redundant.